pkg/apis/core/types_shoot.go

```diff
@@ -585,6 +585,14 @@ type VerticalPodAutoscaler struct {
 	// MemoryHistogramDecayHalfLife is the amount of time it takes a historical memory usage sample to lose half of its weight.
 	// (default: 24h)
 	MemoryHistogramDecayHalfLife *metav1.Duration
+	// MemoryAggregationInterval is the length of a single interval, for which the peak memory usage is computed.
+	// (default: 24h)
+	MemoryAggregationInterval *metav1.Duration
+	// MemoryAggregationIntervalCount is the number of consecutive memory-aggregation-intervals which make up the
+	// MemoryAggregationWindowLength which in turn is the period for memory usage aggregation by VPA. In other words,
+	// `MemoryAggregationWindowLength = memory-aggregation-interval * memory-aggregation-interval-count`.
+	// (default: 8)
+	MemoryAggregationIntervalCount *int64
```

Comment:
You could add validation for the new fields. For example that negative duration cannot be provided for `memoryAggregationInterval ` and `memoryAggregationIntervalCount` cannot be negative as well.

Reply:
I agree that validation would be a useful thing to have for our fields. Do you have an example how/where we do these? Ideally, we would probably use CRD based validations with CEL at some point, right?

Reply:
> Do you have an example how/where we do these?

Yes. You could add the validation in https://github.com/gardener/gardener/blob/b6b857890afabfa10c75a2016f3f02ddde5762a6/pkg/apis/core/validation/shoot.go#L1199. 

For reference, I am also leaving a link to the [Changing the API docs](https://github.com/gardener/gardener/blob/ee88ac263a9bf7c36e5240de1ceddddb065be1c0/docs/development/changing-the-api.md#L19) - it contains the steps to follow when adding a new field to the API. I personally make use of these steps when adding a new field.

---

For CEL, I cannot share much as I haven't looked into the upstream feature. You could open an enhancement issue so that we evaluate it. AFAIK, it is a relatively a new feature.

Reply:
Thanks for the pointer to the docs, sorry I missed this.
Added an issue for converting validations to CEL: https://github.com/gardener/gardener/issues/11266